activesupport/lib/active_support/cache/mem_cache_store.rb

```diff
@@ -120,12 +127,15 @@ def initialize(*addresses)
           raise ArgumentError, "First argument must be an empty array, an array of hosts or a Dalli::Client instance."
         end
         if addresses.first.is_a?(Dalli::Client)
+          ActiveSupport.deprecator.warn(<<~MSG)
+            Initializing MemCacheStore with a Dalli::Client is deprecated and will be removed in Rails 7.2.
```

Comment:
If it's deprecated in favor of something else that should be mentioned in the deprecation notice. We can't guarantee someone will read the docs.

Reply:
First time I hear of this phenomenon 😅